src/commands/createFunctionApp/FunctionAppCreateStep.ts

```diff
@@ -56,7 +57,16 @@ export class FunctionAppCreateStep extends AzureWizardExecuteStep<IFunctionAppWi
                 context.telemetry.properties.fileLoggingError = parseError(error).message;
             }
         }
-
+        const principalId = nonNullProp(nonNullProp(context.site, 'identity'), 'principalId');
```

Comment:
It's a bit cleaner to do 
```suggestion
        const principalId = nonNullValueAndProp(context.site.identity, 'principalId');
```

and I think it does the same thing.